### PR TITLE
[FX] refactor experimental tracer and adapt it with hf models

### PR DESCRIPTION
**Pull request by YuliangLiu0306**
## 📌 Checklist before creating the PR

- [x] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

part effort of #3160 

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

## 💥 Checklist before requesting a review

- [x] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

**Comment by YuliangLiu0306** [comment]
image

File: tests/test_fx/test_tracer/test_timm_model/test_timm_model.py [hunk 1]
```diff
@@ -42,6 +43,7 @@ def trace_and_compare(model_cls, data, output_transform_fn, meta_args=None):
             f'{model.__class__.__name__} has inconsistent outputs, {fx_output_val} vs {non_fx_output_val}'
 
 
+@pytest.mark.skipif(torch.__version__ < '1.12.0', reason='torch version < 12')
```
Comment (FrankLeeeee) [inline tests/test_fx/test_tracer/test_timm_model/test_timm_model.py:46]: You cannot compare like this, you should use `packaging.version` to compare instead.

Comment (YuliangLiu0306) [reply tests/test_fx/test_tracer/test_timm_model/test_timm_model.py:46]: okay

File: tests/kit/model_zoo/__init__.py [hunk 2]
```diff
@@ -1,4 +1,9 @@
-from . import diffusers, timm, torchaudio, torchrec, torchvision, transformers
+from . import timm, torchrec, torchvision, transformers
+
+try:
+    from . import diffusers, torchaudio
+except:
+    pass
```
[the remark this answers is not in the document]
Comment (YuliangLiu0306) [reply tests/kit/model_zoo/__init__.py]: Okay

**Review by FrankLeeeee — APPROVED** [review]

**Pull request merged**